tests/test_util.py

```diff
@@ -1,127 +1,143 @@
+from contextlib import suppress
 from unittest import TestCase, mock
 import os
 import os.path as osp
 
-from datumaro.util import (
-    Rollback, error_rollback, is_method_redefined, on_error_do,
-)
+from datumaro.util import is_method_redefined
 from datumaro.util.os_util import walk
+from datumaro.util.scope import Scope, on_error_do, on_exit_do, scoped
 from datumaro.util.test_utils import TestDir
 
 from .requirements import Requirements, mark_requirement
 
 
-class TestRollback(TestCase):
+class TestException(Exception):
+    pass
+
+class TestScope(TestCase):
     @mark_requirement(Requirements.DATUM_GENERAL_REQ)
-    def test_does_not_call_on_no_error(self):
-        success = True
-        def cb():
-            nonlocal success
-            success = False
+    def test_calls_on_no_error(self):
+        error_cb = mock.MagicMock()
+        exit_cb = mock.MagicMock()
 
-        with Rollback() as on_error:
-            on_error.do(cb)
+        with suppress(TestException), Scope() as scope:
```

Comment:
The suppression shouldn't be needed here'

Reply:
Fixed